--- /usr/portage/media-libs/gst-plugins-base/gst-plugins-base-0.10.29.ebuild	2010-09-25 00:44:09.189166941 -0400
+++ ./media-libs/gst-plugins-base/gst-plugins-base-0.10.29.ebuild	2010-09-25 00:45:02.385828356 -0400
@@ -36,7 +36,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	# gst doesnt handle opts well, last tested with 0.10.15
 	strip-flags
 	replace-flags "-O3" "-O2"
@@ -44,7 +44,6 @@
 	gst-plugins-base_src_configure \
 		--disable-introspection \
 		$(use_enable nls)
-	emake || die "emake failed."
 }
 
 src_install() {
