--- /usr/portage/dev-util/pkgconfig/pkgconfig-0.25-r2.ebuild	2010-07-11 07:28:28.000000000 -0500
+++ ./dev-util/pkgconfig/pkgconfig-0.25-r2.ebuild	2010-07-12 01:24:50.705031212 -0500
@@ -33,6 +33,19 @@
 	# lines are inherited between libraries
 	local myconf
 	use elibc_FreeBSD && myconf="--enable-indirect-deps"
+ 
+	# adjust the default pc search path
+	if [[ -n EMULTILIB_PKG ]]; then
+		local pc_path="/usr/$(get_libdir)/pkgconfig"
+		local abi
+		for abi in ${MULTILIB_ABIS}; do
+			if [[ "$(get_libdir)" != "$(get_abi_LIBDIR ${abi})" ]]; then
+				pc_path="${pc_path}:/usr/$(get_abi_LIBDIR ${abi})/pkgconfig"
+			fi
+		done
+		pc_path="${pc_path}:/usr/share/pkgconfig"
+		myconf="${myconf} --with-pc-path=${pc_path}"
+	fi
 
 	econf \
 		--docdir=/usr/share/doc/${PF}/html \
