--- /auto/portage/dev-util/pkgconfig/pkgconfig-0.23.ebuild	2009-04-05 06:37:41.000000000 -0400
+++ ./dev-util/pkgconfig/pkgconfig-0.23.ebuild	2010-11-23 12:15:26.292250288 -0500
@@ -20,7 +20,7 @@
 
 S=${WORKDIR}/${MY_P}
 
-src_compile() {
+src_configure() {
 	local myconf
 
 	use ppc64 && use hardened && replace-flags -O[2-3] -O1
@@ -29,8 +29,20 @@
 	# lines are inherited between libraries
 	use elibc_FreeBSD && myconf="${myconf} --enable-indirect-deps"
 
+	# adjust the default pc search path
+	if [[ -n EMULTILIB_PKG ]]; then
+		local pc_path="/usr/$(get_libdir)/pkgconfig"
+		local abi
+		for abi in ${MULTILIB_ABIS}; do
+			if [[ "$(get_libdir)" != "$(get_abi_LIBDIR ${abi})" ]]; then
+				pc_path="${pc_path}:/usr/$(get_abi_LIBDIR ${abi})/pkgconfig"
+			fi
+		done
+		pc_path="${pc_path}:/usr/share/pkgconfig"
+		myconf="${myconf} --with-pc-path=${pc_path}"
+	fi
+
 	econf ${myconf} || die "econf failed"
-	emake || die "emake failed"
 }
 
 src_install() {
