--- /auto/portage/dev-libs/libatomic_ops/libatomic_ops-1.2-r1.ebuild	2010-04-26 15:05:37.000000000 -0400
+++ ./dev-libs/libatomic_ops/libatomic_ops-1.2-r1.ebuild	2010-11-22 23:17:00.310253738 -0500
@@ -16,9 +16,7 @@
 DEPEND=""
 RDEPEND=""
 
-src_unpack(){
-	unpack ${A}
-	cd "${S}"
+src_prepare(){
 	epatch "${FILESDIR}"/${P}-ppc64-load_acquire.patch
 	epatch "${FILESDIR}"/${P}-ppc-asm.patch
 	epatch "${FILESDIR}"/${P}-sh4.patch
