--- /usr/portage/sys-auth/polkit/polkit-0.96-r1.ebuild	2010-07-20 10:37:31.000000000 -0500
+++ ./sys-auth/polkit/polkit-0.96-r1.ebuild	2010-07-23 05:00:00.000244157 -0500
@@ -16,3 +16,3 @@
 KEYWORDS="~alpha amd64 ~arm hppa ~ia64 ~ppc ~ppc64 ~sh ~sparc x86 ~x86-fbsd"
-IUSE="debug doc examples expat nls pam"
+IUSE="debug doc examples expat nls pam introspection"
 # introspection
@@ -24,3 +24,4 @@
 	pam? ( virtual/pam )
-	expat? ( dev-libs/expat )"
+	expat? ( dev-libs/expat )
+	introspection? ( dev-libs/gobject-introspection )"
 DEPEND="${RDEPEND}
@@ -67,3 +68,2 @@
 	econf ${conf} \
-		--disable-introspection \
 		--disable-ansi \
@@ -79,4 +79,4 @@
 		$(use_enable doc gtk-doc) \
-		$(use_enable nls)
-		#$(use_enable introspection)
+		$(use_enable nls) \
+		$(use_enable introspection)
 }
