--- /usr/portage/dev-lang/python/python-2.5.4-r4.ebuild	2010-07-04 16:36:35.000000000 -0500
+++ ./dev-lang/python/python-2.5.4-r4.ebuild	2010-07-07 04:37:04.815903244 -0500
@@ -45,6 +45,7 @@
 			tk? ( >=dev-lang/tk-8.0 )
 			xml? ( >=dev-libs/expat-2 )
 		)
+		app-arch/bzip2
 		doc? ( dev-python/python-docs:${SLOT} )"
 DEPEND="${RDEPEND}
 		dev-util/pkgconfig"
@@ -69,10 +70,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	# Ensure that internal copies of expat, libffi and zlib are not used.
 	rm -fr Modules/expat
 	rm -fr Modules/_ctypes/libffi*
@@ -187,11 +185,6 @@
 		--with-system-ffi
 }
 
-src_compile() {
-	src_configure
-	emake || die "emake failed"
-}
-
 src_test() {
 	# Tests will not work when cross compiling.
 	if tc-is-cross-compiler; then
