--- /auto/portage/sys-fs/fuse/fuse-2.7.0.ebuild	2009-04-05 04:08:24.000000000 -0400
+++ ./sys-fs/fuse/fuse-2.7.0.ebuild	2010-11-23 13:48:43.641690589 -0500
@@ -37,18 +37,19 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/fuse-fix-lazy-binding.patch
 	elibtoolize
 }
 
-src_compile() {
+src_configure() {
 	econf \
 		--disable-kernel-module \
 		--disable-example \
 		|| die "econf failed for fuse userland"
+}
+
+src_compile() {
 	emake || die "emake failed"
 
 	if use kernel_linux ; then
