--- /usr/portage/net-misc/curl/curl-7.20.0-r2.ebuild	2010-07-18 00:07:03.000000000 -0500
+++ ./net-misc/curl/curl-7.20.0-r2.ebuild	2010-07-23 05:14:49.282165459 -0500
@@ -47,5 +47,3 @@
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${P}-strip-ldflags.patch
@@ -55,4 +53,3 @@
 
-src_compile() {
-
+src_configure() {
 	myconf="$(use_enable ldap)
@@ -98,4 +95,2 @@
 	econf ${myconf} || die 'configure failed'
-
-	emake || die "install failed for current version"
 }
@@ -113,2 +108,4 @@
 	dodoc docs/MANUAL docs/FAQ docs/BUGS docs/CONTRIBUTE
+
+	prep_ml_includes /usr/include/curl
 }
