--- /auto/portage/x11-libs/openmotif/openmotif-2.2.3-r10.ebuild	2010-10-12 04:37:51.000000000 -0400
+++ ./x11-libs/openmotif/openmotif-2.2.3-r10.ebuild	2010-11-22 18:09:53.212097040 -0500
@@ -28,10 +28,7 @@
 
 S=${WORKDIR}/${MY_P}
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	EPATCH_SUFFIX=patch epatch
 
 	# This replaces deprecated, obsoleted and now invalid AC_DEFINE
@@ -46,7 +43,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	# get around some LANG problems in make (#15119)
 	unset LANG
 
@@ -60,7 +57,6 @@
 	append-flags -fno-strict-aliasing
 
 	econf --with-x || die "econf failed"
-	emake -j1 || die "emake failed"
 }
 
 src_install() {
