--- /auto/portage/net-dns/libidn/libidn-1.19.ebuild	2010-07-07 12:36:42.000000000 -0400
+++ ./net-dns/libidn/libidn-1.19.ebuild	2010-07-09 00:05:01.312533868 -0400
@@ -30,7 +30,7 @@
 	rm "${S}/java/${P}.jar" || die
 }
 
-src_compile() {
+src_configure() {
 	econf \
 		$(use_enable nls) \
 		$(use_enable java) \
@@ -41,7 +41,9 @@
 		--with-packager-version="r${PR}" \
 		--with-packager-bug-reports="https://bugs.gentoo.org" \
 		|| die
+}
 
+src_compile() {
 	emake || die
 
 	if use emacs; then
