--- /auto/portage/dev-libs/gmp/gmp-4.3.2.ebuild	2010-11-17 11:35:35.000000000 -0500
+++ ./dev-libs/gmp/gmp-4.3.2.ebuild	2010-11-22 22:46:47.183768872 -0500
@@ -14,9 +14,7 @@
 KEYWORDS="alpha amd64 arm hppa ia64 m68k ~mips ppc ppc64 s390 sh sparc x86 ~sparc-fbsd ~x86-fbsd"
 IUSE="nocxx" #doc
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	[[ -d ${FILESDIR}/${PV} ]] && EPATCH_SUFFIX="diff" EPATCH_FORCE="yes" epatch "${FILESDIR}"/${PV}
 	epatch "${FILESDIR}"/${PN}-4.1.4-noexecstack.patch
 	epatch "${FILESDIR}"/${PN}-4.3.2-ABI-multilib.patch
@@ -29,7 +27,7 @@
 	elibtoolize
 }
 
-src_compile() {
+src_configure() {
 	# Because of our 32-bit userland, 1.0 is the only HPPA ABI that works
 	# http://gmplib.org/manual/ABI-and-ISA.html#ABI-and-ISA (bug #344613)
 	if [[ ${CHOST} == hppa2.0-* ]] ; then
@@ -51,7 +49,6 @@
 		$(use_enable !nocxx cxx) \
 		|| die "configure failed"
 
-	emake || die "emake failed"
 }
 
 src_install() {
