--- /usr/portage/eclass/toolchain.eclass	2010-11-21 15:35:45.000000000 -0600
+++ ./eclass/toolchain.eclass	2010-12-02 04:53:05.861635599 -0600
@@ -163,4 +163,5 @@ else
 			tc_version_is_at_least "4.1" && IUSE+=" objc++"
 			tc_version_is_at_least "4.2" && IUSE+=" openmp"
+			tc_version_is_at_least "4.2" && IUSE+=" nsplugin"
 			tc_version_is_at_least "4.3" && IUSE+=" fixed-point"
 			if tc_version_is_at_least "4.4" ; then
@@ -1117,5 +1118,5 @@ gcc_src_unpack() {
 	fi
 
-	disable_multilib_libjava || die "failed to disable multilib java"
+#	disable_multilib_libjava || die "failed to disable multilib java"
 }
 
@@ -1208,5 +1209,12 @@ gcc-compiler-configure() {
 		confgcc="${confgcc} --disable-libgcj"
 	elif use gtk ; then
-		confgcc="${confgcc} --enable-java-awt=gtk"
+		confgcc="${confgcc} --enable-java-awt=gtk \
+		$(use_enable nsplugin plugin)"
+	elif use qt ; then
+		confgcc="${confgcc} --enable-java-awt=qt"
+	fi
+
+	if is_crosscompile && is_gcj ; then
+		confgcc="${confgcc} --disable-jni"
 	fi
 
@@ -1945,8 +1953,22 @@ gcc_slot_java() {
 
 	# SLOT up libgcj.pc (and let gcc-config worry about links)
-	local libgcj=$(find "${D}"${PREFIX}/lib/pkgconfig/ -name 'libgcj*.pc')
-	if [[ -n ${libgcj} ]] ; then
-		sed -i "/^libdir=/s:=.*:=${LIBPATH}:" "${libgcj}"
-		mv "${libgcj}" "${D}"/usr/lib/pkgconfig/libgcj-${GCC_PV}.pc || die
+	local libgcj
+	for libgcj in "${D}${PREFIX}"/lib/pkgconfig/libgcj*.pc ; do
+		if [[ -f ${libgcj} ]] ; then
+			sed -i "/^libdir=/s:=.*:=${LIBPATH}:" "${libgcj}"
+			if ! is_crosscompile; then
+				mv "${libgcj}" "${D}"/${PREFIX}/lib/pkgconfig/libgcj-${GCC_PV}.pc || die
+			else
+				dodir /${PREFIX}/${CTARGET}/${PREFIX}/lib/pkgconfig
+				mv "${libgcj}" "${D}"/${PREFIX}/${CTARGET}/${PREFIX}/lib/pkgconfig/libgcj-${GCC_PV}.pc || die
+			fi
+		fi
+	done
+
+	# Move JNI native (CTARGET) java libraries  
+	if is_crosscompile; then
+		dodir /${PREFIX}/${CTARGET}/${PREFIX}/lib
+		find "${D}"/${PREFIX}/lib -name 'gcj-*' -type d -exec \
+			mv {} "${D}"/${PREFIX}/${CTARGET}/${PREFIX}/lib/ \;
 	fi
 
