--- /usr/portage/dev-libs/openssl/openssl-0.9.8o-r1.ebuild	2010-07-20 07:05:40.000000000 -0500
+++ ./dev-libs/openssl/openssl-0.9.8o-r1.ebuild	2010-07-23 04:43:25.666252047 -0500
@@ -34,6 +34,3 @@
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-0.9.7e-gentoo.patch
@@ -70,3 +67,3 @@
 
-src_compile() {
+src_configure() {
 	unset APPS #197996
@@ -124,3 +121,5 @@
 		Makefile || die
+}
 
+src_compile() {
 	# depend is needed to use $confopts
