--- /usr/portage/dev-libs/openssl/openssl-1.0.0a-r1.ebuild	2010-08-10 16:04:59.000000000 -0500
+++ ./dev-libs/openssl/openssl-1.0.0a-r1.ebuild	2010-08-11 10:28:54.993506445 -0500
@@ -24,6 +24,3 @@
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-0.9.7e-gentoo.patch
@@ -60,3 +57,3 @@
 
-src_compile() {
+src_configure() {
 	unset APPS #197996
@@ -115,3 +112,5 @@
 		Makefile || die
+}
 
+src_compile() {
 	# depend is needed to use $confopts
