--- /usr/portage/dev-libs/openssl/openssl-1.0.0a.ebuild	2010-08-03 22:08:25.000000000 -0500
+++ ./dev-libs/openssl/openssl-1.0.0a.ebuild	2010-08-06 07:02:34.988271954 -0500
@@ -24,6 +24,3 @@
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-0.9.7e-gentoo.patch
@@ -58,3 +55,3 @@
 
-src_compile() {
+src_configure() {
 	unset APPS #197996
@@ -113,3 +110,5 @@
 		Makefile || die
+}
 
+src_compile() {
 	# depend is needed to use $confopts
