--- /usr/portage/dev-libs/openssl/openssl-1.0.0a.ebuild	2010-07-18 15:05:39.000000000 -0500
+++ ./dev-libs/openssl/openssl-1.0.0a.ebuild	2010-07-23 04:45:26.546912658 -0500
@@ -24,6 +24,3 @@
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-0.9.7e-gentoo.patch
@@ -57,3 +54,3 @@
 
-src_compile() {
+src_configure() {
 	unset APPS #197996
@@ -112,3 +109,5 @@
 		Makefile || die
+}
 
+src_compile() {
 	# depend is needed to use $confopts
