--- /auto/portage/media-sound/wavpack/wavpack-4.50.1.ebuild	2009-04-05 06:37:41.000000000 -0400
+++ ./media-sound/wavpack/wavpack-4.50.1.ebuild	2010-11-23 13:13:11.895237701 -0500
@@ -13,16 +13,12 @@
 KEYWORDS="alpha amd64 arm hppa ia64 ~mips ppc ppc64 sparc x86 ~x86-fbsd"
 IUSE="mmx"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
 	elibtoolize
 }
 
-src_compile() {
+src_configure() {
 	econf $(use_enable mmx)
-	emake || die "emake failed."
 }
 
 src_install() {
