--- /auto/portage/dev-libs/libsigc++/libsigc++-2.2.3.ebuild	2009-02-27 09:35:50.000000000 -0500
+++ ./dev-libs/libsigc++/libsigc++-2.2.3.ebuild	2010-11-23 10:37:16.079506646 -0500
@@ -12,10 +12,8 @@
 KEYWORDS="alpha amd64 arm hppa ia64 ~mips ppc ppc64 s390 sh sparc x86 ~x86-fbsd"
 IUSE="debug doc test"
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-
+src_prepare() {
+	
 	# don't waste time building examples/docs
 	sed -i 's|^\(SUBDIRS =.*\)docs examples\(.*\)$|\1\2|' Makefile.in || \
 		die "sed docs/examples failed"
@@ -33,7 +31,7 @@
 	fi
 }
 
-src_compile() {
+src_configure() {
 	filter-flags -fno-exceptions
 
 	local myconf
@@ -42,7 +40,6 @@
 		|| myconf="--enable-debug=no"
 
 	econf ${myconf} || die "econf failed."
-	emake || die "emake failed"
 }
 
 src_install() {
