--- /usr/portage/sys-fs/udev/udev-159.ebuild	2010-08-02 16:15:45.000000000 -0500
+++ ./sys-fs/udev/udev-159.ebuild	2010-08-06 06:59:22.442006972 -0500
@@ -141,5 +141,5 @@
 	fi
+}
 
-	cd "${S}"
-
+src_prepare() {
 	# patches go here...
@@ -185,3 +185,3 @@
 
-src_compile() {
+src_configure() {
 	filter-flags -fprefetch-loop-arrays
@@ -201,4 +201,2 @@
 	# we don't have gobject-introspection in portage tree
-
-	emake || die "compiling udev failed"
 }
