--- /usr/portage/sys-fs/udev/udev-149.ebuild	2010-08-02 16:15:45.000000000 -0500
+++ ./sys-fs/udev/udev-149.ebuild	2010-08-06 06:58:45.380841245 -0500
@@ -26,3 +26,3 @@
 KEYWORDS="~alpha amd64 arm ~hppa ~ia64 ~m68k ~mips ppc ppc64 ~s390 -sh ~sparc x86"
-IUSE="selinux +devfs-compat -extras test"
+IUSE="selinux +devfs-compat -extras test introspection"
 
@@ -38,3 +38,4 @@
 	>=sys-apps/util-linux-2.16
-	>=sys-libs/glibc-2.9"
+	>=sys-libs/glibc-2.9
+	introspection? ( >=dev-libs/gobject-introspection-0.6.5 )"
 
@@ -140,5 +141,5 @@
 	fi
+}
 
-	cd "${S}"
-
+src_prepare() {
 	# patches go here...
@@ -185,3 +186,3 @@
 
-src_compile() {
+src_configure() {
 	filter-flags -fprefetch-loop-arrays
@@ -198,6 +199,3 @@
 		$(use_enable extras) \
-		--disable-introspection
-	# we don't have gobject-introspection in portage tree
-
-	emake || die "compiling udev failed"
+		$(use_enable introspection)
 }
@@ -312,2 +310,3 @@
 	dohtml *.html
+	cd "${S}"
 }
