--- /usr/portage/net-libs/xulrunner/xulrunner-1.9.2.7-r1.ebuild	2010-07-23 07:45:10.000000000 -0500
+++ ./net-libs/xulrunner/xulrunner-1.9.2.7-r1.ebuild	2010-07-24 02:25:56.680018182 -0500
@@ -223,3 +223,3 @@
 	dodir /etc/env.d
-	echo "LDPATH=${EPREFIX}/${MOZLIBDIR}" > "${ED}"/etc/env.d/08xulrunner || die "env.d failed"
+	echo "LDPATH=${EPREFIX}/${MOZLIBDIR}" > "${ED}"/etc/env.d/08xulrunner-${CHOST} || die "env.d failed"
 
@@ -239,2 +239,3 @@
 pkg_postinst() {
+	if is_final_abi ; then
 	ewarn "If firefox fails to start with \"failed to load xpcom\", run revdep-rebuild"
@@ -252,2 +253,3 @@
 	ewarn "should be cc'd on the bug report. Thank you anarchy@gentoo.org ."
+	fi
 }
