--- /auto/portage/dev-lang/tcl/tcl-8.5.7.ebuild	2010-06-15 20:05:39.000000000 -0400
+++ ./dev-lang/tcl/tcl-8.5.7.ebuild	2010-11-22 22:28:53.170542534 -0500
@@ -31,9 +31,7 @@
 	fi
 }
 
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
+src_prepare() {
 	epatch "${FILESDIR}"/${PN}-8.5_alpha6-multilib.patch
 
 	# Bug 125971
@@ -43,7 +41,7 @@
 	eautoreconf
 }
 
-src_compile() {
+src_configure() {
 	# workaround stack check issues, bug #280934
 	if use hppa; then
 		append-cflags "-DTCL_NO_STACK_CHECK=1"
@@ -55,7 +53,6 @@
 	econf \
 		$(use_enable threads) \
 		$(use_enable debug symbols) || die
-	emake || die
 }
 
 src_install() {
