--- /usr/portage/x11-libs/pango/pango-1.26.2.ebuild	2010-07-21 06:39:09.000000000 -0500
+++ ./x11-libs/pango/pango-1.26.2.ebuild	2010-07-23 05:12:58.869965181 -0500
@@ -15,3 +15,3 @@
 KEYWORDS="~alpha amd64 ~arm hppa ~ia64 ~mips ~ppc ppc64 ~s390 ~sh ~sparc x86 ~x86-fbsd"
-IUSE="X doc test"
+IUSE="X doc test introspection"
 
@@ -24,3 +24,4 @@
 		x11-libs/libX11
-		x11-libs/libXft )"
+		x11-libs/libXft )
+	introspection? ( dev-libs/gobject-introspection )"
 DEPEND="${RDEPEND}
@@ -48,3 +49,3 @@
 	G2CONF="${G2CONF}
-		--disable-introspection
+		$(use_enable introspection)
 		$(use_with X x)"
