--- /auto/portage/dev-db/mysql/mysql-5.1.52.ebuild	2010-11-06 15:37:14.000000000 -0400
+++ ./dev-db/mysql/mysql-5.1.52.ebuild	2010-11-22 22:33:50.303846170 -0500
@@ -26,7 +26,7 @@
 DEPEND="|| ( >=sys-devel/gcc-3.4.6 >=sys-devel/gcc-apple-4.0 )
 		>=sys-devel/libtool-2.2.10"
 
-# Please do not add a naive src_unpack to this ebuild
+# Please do not add a naive src unpack to this ebuild
 # If you want to add a single patch, copy the ebuild to an overlay
 # and create your own mysql-extras tarball, looking at 000_index.txt
 src_prepare() {
